### ISSUE TITLE
Debounce file saving

### ISSUE THREAD

**Issue by ishaan1013** [ishaan1013/sandbox#13]
We currently update Cloudflare R2 on every file save.

Implementing debouncing would reduce # of queries when a user frequently saves the same file.

**Comment by akhilkh2000** [comment]
@ishaan1013  - Have raised a PR for this, let me know if that looks good. Also, had noticed a couple of issues / observations that we could discuss that can be raised as separate issues:

1. I see that each key press gets sent to liveblocks so , we might need debouncing around that as well (unless thats a default unchangeable behavior of liveblocks itself) 
2.  Playing around with the editor and typing in it results in thousands of socket XHR requests , not sure if that is by design 
![image](https://github.com/ishaan1013/sandbox/assets/56243241/15a6cbd6-111c-472e-bd0a-b138b0603698)
****

After sometime I see these errors on the console
![image](https://github.com/ishaan1013/sandbox/assets/56243241/4744e50d-dfd5-4297-aeaf-d907d7c9cb71)

3. Also, one error I ran into while testing my change was if "undefined" is passed as the file content to the backend , it returns a rate limiter error as the catch all exception block just sends that by default which might be a little misleading as its not actually a rate limit issue 
![image](https://github.com/ishaan1013/sandbox/assets/56243241/5cc6db5b-2e78-470c-b40e-493f925867fb)

**Comment by akhilkh2000** [comment]
Also, Im not very well versed with CF but in about a day, looking at the metrics, it looks like we can optimize the number of requests to CF (other than the debouncing) as well? 
![image](https://github.com/ishaan1013/sandbox/assets/56243241/55be4e4d-525e-4d5e-8b8a-ab1c70868ccf)

**Comment by ishaan1013** [comment]
Great points. In all honesty I didn't do a ton of testing and I noticed a few of the issues you mentioned, although not specifically to this extent. There's definitely a ton of room for optimization, so feel free to continue looking into it and keep me updated! I'd love to hear about your journey through this and help you out along the way.

**Comment by akhilkh2000** [comment]
@ishaan1013 yeah, will try looking at it sometime next week. Btw, does the pr look good to merge? Let me know if there is any issues with it
